src/HLLD.hpp

```diff
@@ -0,0 +1,343 @@
+#ifndef HLLD_HPP_ // NOLINT
+#define HLLD_HPP_
+
+#include "AMReX_Extension.H"
+#include "AMReX_GpuQualifiers.H"
+#include <AMReX.H>
+#include <AMReX_REAL.H>
+
+#include "ArrayView.hpp"
+#include "HydroState.hpp"
+#include "valarray.hpp"
+
+namespace quokka::Riemann
+{
+#define DELTA 1.0e-4
+
+template <class T> constexpr auto SQUARE(const T x) -> T { return x * x; }
+
+template <int N_scalars> auto FastMagnetosonicSpeed(double gamma, quokka::HydroState<N_scalars> const state, const double bx) -> double
```

Comment:
This will not compile on GPU as-is. You need to add `AMREX_FORCE_INLINE AMREX_GPU_DEVICE` annotations to this function, just like the HLLD() function.